src/main/java/org/kohsuke/github/GHRepository.java

```diff
@@ -2744,6 +2761,26 @@ public GHRepositoryVariable getRepoVariable(String name) throws IOException {
                 .fetch(GHRepositoryVariable.class);
     }
 
+    /**
+     * Update repository variable
+     *
+     * @param name
+     *            the name
+     * @param variableToUpdate
+     *            the variable to update
+     * @throws IOException
+     *             the io exception
+     */
+    public void updateRepoVariable(final String name, final GHRepositoryVariable variableToUpdate) throws IOException {
+
+        root().createRequest()
+                .withUrlPath(getApiTailUrl("actions/variables"), name)
+                .method("PATCH")
+                .with("name", variableToUpdate.getName())
+                .with("value", variableToUpdate.getValue())
+                .send();
+    }
+
```

Comment:
As noted above I think this should be moved to `GHRepositoryVariable`.   Even if you don't want to use the pattern that `GHLabel` uses, the update should be done from an instance method on `GHRepositoryVariable`.

Reply:
Of course, the GHLabel pattern sounds good.